Modules/_sqlite/statement.c

```diff
@@ -377,24 +374,15 @@ int pysqlite_statement_reset(pysqlite_Statement* self)
 
     rc = SQLITE_OK;
 
-    if (self->in_use && self->st) {
+    if (self->st) {
```

Comment:
This may lead to `sqlite3_reset` being called multiple times on a statement, but there is ongoing effort to optimise this part of the SQLite API usage ([issue 43350](https://bugs.python.org/issue43350)).